doc/ROS2_PX4_Offboard_Tutorial.md

```diff
@@ -0,0 +1,415 @@
+#All required rosdeps installed successfully
+analyst@casa:~/microros_ws$ ros2 run micro_ros_setup build_agent.sh
+Building micro-ROS Agent
+[0.593s] WARNING:colcon.colcon_ros.prefix_path.ament:The path '/home/analyst/microros_ws/install/micro_ros_agent' in the environment variable AMENT_PREFIX_PATH doesn't exist
+[0.593s] WARNING:colcon.colcon_ros.prefix_path.ament:The path '/home/analyst/microros_ws/install/micro_ros_msgs' in the environment variable AMENT_PREFIX_PATH doesn't exist
+[0.593s] WARNING:colcon.colcon_ros.prefix_path.catkin:The path '/home/analyst/microros_ws/install/micro_ros_agent' in the environment variable CMAKE_PREFIX_PATH doesn't exist
+[0.593s] WARNING:colcon.colcon_ros.prefix_path.catkin:The path '/home/analyst/microros_ws/install/micro_ros_msgs' in the environment variable CMAKE_PREFIX_PATH doesn't exist
+Starting >>> micro_ros_msgs
+Finished <<< micro_ros_msgs [5.51s]                     
+Starting >>> micro_ros_agent
+--- stderr: micro_ros_agent                                
+Cloning into 'xrceagent'...
+Switched to a new branch 'ros2'
+HEAD is now at 3eb56b5 Release v2.3.0
+CMake Warning (dev) at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
+  The package name passed to `find_package_handle_standard_args` (tinyxml2)
+  does not match the name of the calling package (TinyXML2).  This can lead
+  to problems in calling code that expects `find_package` result variables
+  (e.g., `_FOUND`) to follow a certain pattern.
+Call Stack (most recent call first):
+  cmake/modules/FindTinyXML2.cmake:40 (find_package_handle_standard_args)
+  /opt/ros/humble/share/fastrtps/cmake/fastrtps-config.cmake:51 (find_package)
+  CMakeLists.txt:153 (find_package)
+This warning is for project developers.  Use -Wno-dev to suppress it.
+
+---
+Finished <<< micro_ros_agent [23.2s]
+
+Summary: 2 packages finished [29.0s]
+  1 package had stderr output: micro_ros_agent
+
+```
+
+Try running the agent:
+
+```
+source install/local_setup.sh
+ros2 run micro_ros_agent micro_ros_agent udp4 --port 8888
+```
+
+It should respond with:
+
+```
+[1670101284.566587] info     | UDPv4AgentLinux.cpp | init                     | running...             | port: 8888
+[1670101284.566822] info     | Root.cpp           | set_verbose_level        | logger setup           | verbose_level: 4
+```
+
+## Run the Demo
+
+### Overview
+
+You will need 4 terminal windows  ([gnome terminator](https://gnome-terminator.org/) is a great for this!), one for each of the following components:
+
+   * micro-ros-agent
+   * Gazebo   
+   * px4-offboard example
+   * QGroundControl
+
+In each terminal window, we will first source any workspace settings required for a particular component and also set the `ROS_DOMAIN_ID` and `PYTHONOPTIMIZE`  in each window
+
+### Start the micro-ros-agent
+
+In the terminal you designated as the `micro-ros-agent` terminal:
+
+```
+cd ~/microros_ws
+export ROS_DOMAIN_ID=0
+export PYTHONOPTIMIZE=1
```

Comment:
Is this necessary?

Reply:
Yes.  As a newbie, I struggled for hours before I realized that my `ROS_DOMAIN_ID` had to be consistent across the 3 processes.  PYTHONOPTIMIZE is required in my environment for GAZEBO to work.

Reply:
Do you know why PYTHONOPTIMIZE is required? what environment are you on?

Reply:
My environment:

```
   * Ubuntu 22.02
   * ROS2 Humble
   * Python 3.10
```

When I ran through this the first time, months ago, I got this error:

![image](https://user-images.githubusercontent.com/2936416/229311006-cf69a76b-a898-4979-ac0e-327e600a7bde.png)

After several hours, I found the solution [here](https://github.com/PX4/px4_ros_com/issues/162)

Reply:
Okay, strange